<a href="https://colab.research.google.com/github/sk-aili/MyHub/blob/main/Hands_On_Approach_to_Causal_Inference.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Business Problem: The Sprinkler Problem    

How effective is our sprinkler is making the grass wet?   
Treatment: Sprinkler - 1 (Turned On) or 0 (Turned Off)    
Outcome: Wet grass - 1 (Wet) or 0 (Dry)    

Other Variables:    
Rain - 1 (Rain) or 0 (Didn't rain)    
Cloudy - 1 (Cloudy) or 0 (wasn't Cloudy)

## Prerequisites

Install Requirements

In [1]:
! pip install numpy==1.24.2

In [4]:
! pip install bnlearn --quiet

Import libraries

In [3]:
import pandas as pd
import numpy as np
import bnlearn as bn

## Dataset

In [6]:
df = bn.import_example("sprinkler")
df.head()

Cloudy  Sprinkler  Rain  Wet_Grass
0       0          0     0          0
1       1          0     1          1
2       0          1     0          1
3       1          1     1          1
4       1          1     1          1

## Step 0: How different are our treatment (Sprinkler == 1) and Control (Sprinkler == 0) Groups? (in regards to Cloudy)

In [8]:
df.groupby(['Sprinkler', 'Cloudy']).agg({'Wet_Grass': 'count'})

Wet_Grass
Sprinkler Cloudy           
0         0             225
          1             466
1         0             263
          1              46

We have a lot less cloudy days (46) in our Treatment group as opposed to the control group (466). We need to match them. How to do it? [Using Propensity score]